### ISSUE TITLE
`:config-in-call` for `:aliased-namespace-var-usage` ineffective in quoted forms

### ISSUE THREAD

**Issue by colin-p-hill** [clj-kondo/clj-kondo#1962]
**version**

clj-kondo v2023.01.16

**platform**

Windows 11, native, scoop.

**problem**

Inside quoted forms, such as a typical macro's body, `:config-in-call` does not seem capable of disabling the `:aliased-namespace-var-usage` linter. This can make it difficult to weed out false positives when using macros that can validly accept `:as-alias`-qualified symbols.

**repro**

```clojure
(ns repro
  (:require [clojure.test.check.generators :as-alias gen]))

(defn resolve-alias [sym]
  (let [ns-sym   (symbol (namespace sym))
        sym-name (name sym)]
    (symbol (or (some-> *ns* ns-aliases ns-sym ns-name str)
                (namespace sym))
            sym-name)))

(defmacro res [sym]
  `@(requiring-resolve (resolve-alias '~sym)))

(defmacro sample [sym]
  `((res gen/sample) (res ~sym))) ; warning here

`(res gen/string)                 ; warning here

(sample gen/string)
```
```text
> clj-kondo --lint .
.\repro.clj:15:10: warning: Namespace only aliased but wasn't loaded: clojure.test.check.generators
.\repro.clj:17:7: warning: Namespace only aliased but wasn't loaded: clojure.test.check.generators
linting took 34ms, errors: 0, warnings: 2
```

**config**

```edn
{:config-in-call {repro/res    {:linters {:aliased-namespace-var-usage {:level :off}}}
                  repro/sample {:linters {:aliased-namespace-var-usage {:level :off}}}}}
```

**expected behavior**

I expect the configuration provided above to disable these warnings.

**Comment by borkdude** [comment]
Yeah, this should be disabled by default, using vars like that in syntax-quote is valid without loading the namespace
